code/getkey.c

```diff
@@ -0,0 +1,70 @@
+#include <ctype.h>
```

Comment:
good practice to always include all the header files and user defined header in .h files and write just code in .c file.            

Remove all the header files and just include getkey.h

Reply:
As I see it, in a .h file you should only add .h files that are directly related to the header file you are defining. 
In the example below the stddef.h library is needed because it is used for type size_t. 
The rest of the libraries should be declared in the .c file. 
The same thing happens with functions, just write the prototypes of the functions that are going to be used outside the .c file. 
At least that's what I've read as good practices declaring .c and .h files

```c
#ifndef MY_LIBRARY_H
#define MY_LIBRARY_H

#include <stddef.h>  // Need to use size_t

#define MAX_SIZE 100

size_t get_array_size(int arr[], size_t length);

#endif // MY_LIBRARY_H

Reply:
Yes you are right This is true when you are writing system specific code. 

But here as we are developing cross platform program to work.. including all the header in .c file will unnecessarily including extra header file like if i am running the game in windows i don't need (#include<ctype.h>  #include <stdio.h>) As similarly if i am in Linux system in would not need #include<windows.h> .. 

So, i have make the declarartion including functions declarations into header file.. So that it only includes what it needs